Adopt ASCIILiterals in makeString, tryMakeString, operator+ calls inside JavaScriptCore

zdobersek opened the pull request:
#### cb3a9da0118c397a5d2eb134f6cdc32509e93147
<pre>
Adopt ASCIILiterals in makeString, tryMakeString, operator+ calls inside JavaScriptCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=252167">https://bugs.webkit.org/show_bug.cgi?id=252167</a>

Reviewed by Darin Adler.

Convert different cases of makeString(), tryMakeString() and StringAppend-producing
operator+ calls from using native C string literals to using more-desired ASCIILiterals.

* Source/JavaScriptCore/API/glib/JSCCallbackFunction.cpp:
(JSC::JSCCallbackFunction::construct):
* Source/JavaScriptCore/API/glib/JSCContext.cpp:
(jscContextGArrayToJSArray):
(jscContextJSArrayToGArray):
(jscContextGValueToJSValue):
(jscContextJSValueToGValue):
* Source/JavaScriptCore/API/glib/JSCValue.cpp:
(jsc_value_new_array):
(jscValueCallFunction):
* Source/JavaScriptCore/API/tests/ExecutionTimeLimitTest.cpp:
(testExecutionTimeLimit):
* Source/JavaScriptCore/API/tests/testapi.cpp:
(TestAPI::markedJSValueArrayAndGC):
* Source/JavaScriptCore/assembler/LinkBuffer.cpp:
(JSC::LinkBuffer::linkComments):
* Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp:
(JSC::DFG::SpeculativeJIT::runSlowPathGenerators):
(JSC::DFG::SpeculativeJIT::compileCurrentBlock):
* Source/JavaScriptCore/ftl/FTLLowerDFGToB3.cpp:
(JSC::FTL::DFG::LowerDFGToB3::compileCompareStrictEq):
* Source/JavaScriptCore/inspector/IdentifiersFactory.cpp:
(Inspector::addPrefixToIdentifier):
* Source/JavaScriptCore/inspector/InjectedScriptBase.cpp:
(Inspector::InjectedScriptBase::makeCall):
(Inspector::InjectedScriptBase::makeAsyncCall):
* Source/JavaScriptCore/inspector/InspectorBackendDispatcher.cpp:
(Inspector::BackendDispatcher::dispatch):
(Inspector::BackendDispatcher::getPropertyValue):
* Source/JavaScriptCore/inspector/JSGlobalObjectConsoleClient.cpp:
(Inspector::JSGlobalObjectConsoleClient::profile):
(Inspector::JSGlobalObjectConsoleClient::profileEnd):
(Inspector::JSGlobalObjectConsoleClient::warnUnimplemented):
* Source/JavaScriptCore/inspector/agents/InspectorAuditAgent.cpp:
(Inspector::InspectorAuditAgent::run):
* Source/JavaScriptCore/inspector/agents/InspectorConsoleAgent.cpp:
(Inspector::InspectorConsoleAgent::enable):
(Inspector::InspectorConsoleAgent::startTiming):
(Inspector::InspectorConsoleAgent::logTiming):
(Inspector::InspectorConsoleAgent::stopTiming):
(Inspector::InspectorConsoleAgent::count):
(Inspector::InspectorConsoleAgent::countReset):
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp:
(Inspector::objectGroupForBreakpointAction):
(Inspector::InspectorDebuggerAgent::ProtocolBreakpoint::ProtocolBreakpoint):
* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::Interpreter::executeEval):
* Source/JavaScriptCore/jit/JIT.cpp:
(JSC::JIT::privateCompileMainPass):
(JSC::JIT::privateCompileSlowCases):
* Source/JavaScriptCore/jsc.cpp:
(GlobalObject::moduleLoaderImportModule):
(GlobalObject::moduleLoaderResolve):
(GlobalObject::moduleLoaderFetch):
(FunctionJSCStackFunctor::operator() const):
(computeFilePath):
* Source/JavaScriptCore/parser/Lexer.cpp:
(JSC::Lexer&lt;T&gt;::invalidCharacterMessage const):
(JSC::Lexer&lt;LChar&gt;::parseIdentifier):
(JSC::Lexer&lt;T&gt;::scanRegExp):
* Source/JavaScriptCore/parser/Parser.cpp:
(JSC::Parser&lt;LexerType&gt;::parseInner):
* Source/JavaScriptCore/parser/Parser.h:
(JSC::Parser::updateErrorWithNameAndMessage):
* Source/JavaScriptCore/runtime/AbstractModuleRecord.cpp:
(JSC::AbstractModuleRecord::getModuleNamespace):
(JSC::printableName):
* Source/JavaScriptCore/runtime/BytecodeCacheError.cpp:
(JSC::BytecodeCacheError::WriteError::message const):
* Source/JavaScriptCore/runtime/Error.cpp:
(JSC::makeDOMAttributeGetterTypeErrorMessage):
(JSC::makeDOMAttributeSetterTypeErrorMessage):
(JSC::throwConstructorCannotBeCalledAsFunctionTypeError):
(JSC::createOutOfMemoryError):
* Source/JavaScriptCore/runtime/ErrorInstance.cpp:
(JSC::ErrorInstance::sanitizedToString):
* Source/JavaScriptCore/runtime/ExceptionHelpers.cpp:
(JSC::createUndefinedVariableError):
(JSC::defaultApproximateSourceError):
(JSC::defaultSourceAppender):
(JSC::invalidParameterInSourceAppender):
(JSC::invalidParameterInstanceofSourceAppender):
(JSC::invalidPrototypeSourceAppender):
(JSC::createErrorForInvalidGlobalAssignment):
* Source/JavaScriptCore/runtime/FunctionConstructor.cpp:
(JSC::stringifyFunction):
* Source/JavaScriptCore/runtime/IntlCollator.cpp:
(JSC::IntlCollator::initializeCollator):
* Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
(JSC::IntlDateTimeFormat::initializeDateTimeFormat):
* Source/JavaScriptCore/runtime/IntlDurationFormat.cpp:
(JSC::IntlDurationFormat::initializeDurationFormat):
* Source/JavaScriptCore/runtime/IntlNumberFormat.cpp:
(JSC::IntlNumberFormat::initializeNumberFormat):
* Source/JavaScriptCore/runtime/IntlObject.cpp:
(JSC::intlDefaultNumberOption):
(JSC::canonicalizeLocaleList):
* Source/JavaScriptCore/runtime/IntlRelativeTimeFormat.cpp:
(JSC::IntlRelativeTimeFormat::initializeRelativeTimeFormat):
* Source/JavaScriptCore/runtime/JSCJSValueInlines.h:
(JSC::JSValue::toIndex const):
(JSC::JSValue::toTypedArrayIndex const):
* Source/JavaScriptCore/runtime/JSCustomGetterFunction.cpp:
(JSC::JSCustomGetterFunction::create):
* Source/JavaScriptCore/runtime/JSCustomSetterFunction.cpp:
(JSC::JSCustomSetterFunction::create):
* Source/JavaScriptCore/runtime/JSFunction.cpp:
(JSC::makeNameWithOutOfMemoryCheck):
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::init):
* Source/JavaScriptCore/runtime/JSModuleLoader.cpp:
(JSC::JSModuleLoader::importModule):
(JSC::JSModuleLoader::fetch):
* Source/JavaScriptCore/runtime/JSModuleRecord.cpp:
(JSC::JSModuleRecord::instantiateDeclarations):
* Source/JavaScriptCore/runtime/JSObject.cpp:
(JSC::JSObject::putDirectNativeIntrinsicGetter):
(JSC::JSObject::putDirectNativeIntrinsicGetterWithoutTransition):
* Source/JavaScriptCore/runtime/LiteralParser.cpp:
(JSC::LiteralParser&lt;CharType&gt;::Lexer::lex):
(JSC::LiteralParser&lt;CharType&gt;::Lexer::lexStringSlow):
(JSC::LiteralParser&lt;CharType&gt;::parsePrimitiveValue):
* Source/JavaScriptCore/runtime/LiteralParser.h:
(JSC::LiteralParser::getErrorMessage):
* Source/JavaScriptCore/runtime/NumberPrototype.cpp:
(JSC::throwVMToThisNumberError):
* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::Options::setAliasedOption):
* Source/JavaScriptCore/runtime/ProgramExecutable.cpp:
(JSC::ProgramExecutable::initializeGlobalProperties):
* Source/JavaScriptCore/runtime/ProxyObject.cpp:
(JSC::ProxyObject::performPut):
(JSC::ProxyObject::performDefineOwnProperty):
(JSC::ProxyObject::performGetOwnPropertyNames):
* Source/JavaScriptCore/runtime/SamplingProfiler.cpp:
(JSC::SamplingProfiler::reportTopBytecodes):
* Source/JavaScriptCore/runtime/Symbol.cpp:
(JSC::Symbol::descriptiveString const):
* Source/JavaScriptCore/wasm/WasmBBQPlan.cpp:
(JSC::Wasm::BBQPlan::work):
(JSC::Wasm::BBQPlan::compileFunction):
(JSC::Wasm::BBQPlan::didCompleteCompilation):
* Source/JavaScriptCore/wasm/WasmEntryPlan.cpp:
(JSC::Wasm::EntryPlan::prepare):
* Source/JavaScriptCore/wasm/WasmEntryPlan.h:
(JSC::Wasm::EntryPlan::tryReserveCapacity):
* Source/JavaScriptCore/wasm/WasmIndexOrName.cpp:
(JSC::Wasm::makeString):
* Source/JavaScriptCore/wasm/WasmLLIntPlan.cpp:
(JSC::Wasm::LLIntPlan::compileFunction):
(JSC::Wasm::LLIntPlan::didCompleteCompilation):
* Source/JavaScriptCore/wasm/WasmOMGPlan.cpp:
(JSC::Wasm::OMGPlan::work):
* Source/JavaScriptCore/wasm/WasmOSREntryPlan.cpp:
(JSC::Wasm::OSREntryPlan::work):
* Source/JavaScriptCore/wasm/js/WebAssemblyExceptionPrototype.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/wasm/js/WebAssemblyModuleRecord.cpp:
(JSC::WebAssemblyModuleRecord::initializeImports):

Canonical link: <a href="https://commits.webkit.org/261376@main">https://commits.webkit.org/261376@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7660608604b2969cb185fa2b16ac98b127907fed

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111467 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20603 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/99 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/3261 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120251 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21973 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11707 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/2677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117230 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/16313 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99484 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/104197 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98270 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/67 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/45013 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/99996 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13101 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/65 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/86812 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/11214 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13605 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/9493 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/101186 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19050 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52048 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31508 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7910 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15572 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/109224 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26911 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/JavaScriptCore/API/glib/JSCContext.cpp:
```diff
@@ -377,7 +377,7 @@ GUniquePtr<char*> jscContextJSArrayToGStrv(JSCContext* context, JSValueRef jsArr
         auto jsValueItem = jscContextGetOrCreateValue(context, jsItem);
         if (!jsc_value_is_string(jsValueItem.get())) {
-            *exception = toRef(JSC::createTypeError(globalObject, makeString("invalid js type for GStrv: item ", String::number(i), " is not a string")));
+            *exception = toRef(JSC::createTypeError(globalObject, makeString("invalid js type for GStrv: item "_s, String::number(i), " is not a string"_s)));
```
Should say `i` instead of `String::number(i)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/assembler/LinkBuffer.cpp:
```diff
@@ -482,7 +482,7 @@ void LinkBuffer::linkComments(MacroAssembler& assembler)
         String strCopy = str.isolatedCopy();
         if (map.contains(key))
-            strCopy = map.get(key).isolatedCopy() + "\n; " + strCopy;
+            strCopy = map.get(key).isolatedCopy() + "\n; "_s + strCopy;
```
It’s really unclear why we make two isolated copies when we are just going to concatenate and make a new string, which will automatically be “isolated”. Doing it with those copies is unnecessarily expensive and should be fixed.

darinadler commented on Source/JavaScriptCore/ftl/FTLLowerDFGToB3.cpp:
```diff
@@ -19556,7 +19556,7 @@ IGNORE_CLANG_WARNINGS_END
         }
         
-        DFG_CRASH(m_graph, m_node, makeString("Value not defined: ", String::number(edge.node()->index())).ascii().data());
+        DFG_CRASH(m_graph, m_node, makeString("Value not defined: "_s, String::number(edge.node()->index())).ascii().data());
```
Should say `edge.node()->index()` instead of `String::number(edge.node()->index())`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/inspector/InspectorBackendDispatcher.cpp:
```diff
@@ -174,7 +174,7 @@ void BackendDispatcher::dispatch(const String& message)
         SupplementalBackendDispatcher* domainDispatcher = m_dispatchers.get(domain);
         if (!domainDispatcher) {
-            reportProtocolError(MethodNotFound, "'" + domain + "' domain was not found");
+            reportProtocolError(MethodNotFound, "'"_s + domain + "' domain was not found"_s);
```
Using `makeString` would be slightly more efficient because we could use a character rather than a string for the prefix.

darinadler commented on Source/JavaScriptCore/interpreter/Interpreter.cpp:
```diff
@@ -1266,15 +1266,15 @@ JSValue Interpreter::executeEval(EvalExecutable* eval, JSGlobalObject* lexicalGl
             PropertySlot slot(globalLexicalEnvironment, PropertySlot::InternalMethodType::VMInquiry, &vm);
             if (JSGlobalLexicalEnvironment::getOwnPropertySlot(globalLexicalEnvironment, globalObject, ident, slot)) {
-                return throwTypeError(globalObject, throwScope, makeString("Can't create duplicate global variable in eval: '", String(ident.impl()), "'"));
+                return throwTypeError(globalObject, throwScope, makeString("Can't create duplicate global variable in eval: '"_s, String(ident.impl()), '\''));
```
Should be able to pass `ident` to `makeString` without calling `String` and doing reference count churn.

zdobersek replied:
It's a UniquedStringImpl pointer, which would require a separate specialization. That can be added later, but the refcount churn can be avoided already by using StringView.

darinadler commented on Source/JavaScriptCore/interpreter/Interpreter.cpp:
```diff
@@ -1266,15 +1266,15 @@ JSValue Interpreter::executeEval(EvalExecutable* eval, JSGlobalObject* lexicalGl
             PropertySlot slot(globalLexicalEnvironment, PropertySlot::InternalMethodType::VMInquiry, &vm);
             if (JSGlobalLexicalEnvironment::getOwnPropertySlot(globalLexicalEnvironment, globalObject, function->name(), slot)) {
-                return throwTypeError(globalObject, throwScope, makeString("Can't create duplicate global variable in eval: '", String(function->name().impl()), "'"));
+                return throwTypeError(globalObject, throwScope, makeString("Can't create duplicate global variable in eval: '"_s, String(function->name().impl()), '\''));
```
Should be able to pass `function->name()` to `makeString` without calling `String` and doing reference count churn.

darinadler commented on Source/JavaScriptCore/interpreter/Interpreter.cpp:
```diff
@@ -1314,7 +1314,7 @@ JSValue Interpreter::executeEval(EvalExecutable* eval, JSGlobalObject* lexicalGl
-                    return throwSyntaxError(globalObject, throwScope, makeString("Can't create duplicate variable in eval: '", String(function->name().impl()), "'"));
+                    return throwSyntaxError(globalObject, throwScope, makeString("Can't create duplicate variable in eval: '"_s, String(function->name().impl()), '\''));
                 PutPropertySlot slot(variableObject);
                 // We need create this variables because it will be used to emits code by bytecode generator
```
Should be able to pass `function->name()` to `makeString` without calling `String` and doing reference count churn.

darinadler commented on Source/JavaScriptCore/parser/Parser.cpp:
```diff
@@ -306,7 +306,7 @@ Expected<typename Parser<LexerType>::ParseInnerResult, String> Parser<LexerType>
             if (scope->lexicalVariables().contains(impl))
                 return IterationStatus::Continue;
-            errorMessage = makeString("Cannot reference undeclared private names: \"", String(impl), "\"");
+            errorMessage = makeString("Cannot reference undeclared private names: \""_s, String(impl), '"');
```
Should be able to pass `impl` without wrapping in `String` and causing reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/JSModuleLoader.cpp:
```diff
@@ -250,7 +250,7 @@ JSInternalPromise* JSModuleLoader::importModule(JSGlobalObject* globalObject, JS
 
     scope.release();
-    promise->reject(globalObject, createError(globalObject, makeString("Could not import the module '", WTFMove(moduleNameString), "'.")));
+    promise->reject(globalObject, createError(globalObject, makeString("Could not import the module '"_s, WTFMove(moduleNameString), "'."_s)));
```
The `WTFMove` here serves no purpose, and should be removed

darinadler commented on Source/JavaScriptCore/runtime/BytecodeCacheError.cpp:
```diff
@@ -47,7 +47,7 @@ bool BytecodeCacheError::WriteError::isValid() const
 String BytecodeCacheError::WriteError::message() const
 {
-    return makeString("Could not write the full cache file to disk. Only wrote ", String::number(m_written), " of the expected ", String::number(m_expected), " bytes.");
+    return makeString("Could not write the full cache file to disk. Only wrote "_s, String::number(m_written), " of the expected "_s, String::number(m_expected), " bytes."_s);
```
Should say `m_written` and `m_expected` instead of `String::number(m_written)` and  `String::number(m_expected)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmBBQPlan.cpp:
```diff
@@ -161,7 +161,7 @@ void BBQPlan::work(CompilationEffort effort)
     if (UNLIKELY(linkBuffer.didFailToAllocate())) {
         Locker locker { m_lock };
-        Base::fail(makeString("Out of executable memory while tiering up function at index ", String::number(m_functionIndex)));
+        Base::fail(makeString("Out of executable memory while tiering up function at index "_s, String::number(m_functionIndex)));
```
Should say `m_functionIndex` instead of `String::number(m_functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmBBQPlan.cpp:
```diff
@@ -277,7 +277,7 @@ std::unique_ptr<InternalFunction> BBQPlan::compileFunction(uint32_t functionInde
         if (!m_errorMessage) {
             // Multiple compiles could fail simultaneously. We arbitrarily choose the first.
-            fail(makeString(parseAndCompileResult.error(), ", in function at index ", String::number(functionIndex))); // FIXME make this an Expected.
+            fail(makeString(parseAndCompileResult.error(), ", in function at index "_s, String::number(functionIndex))); // FIXME make this an Expected.
```
Should say `functionIndex` instead of `String::number(functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmBBQPlan.cpp:
```diff
@@ -297,7 +297,7 @@ void BBQPlan::didCompleteCompilation()
             InternalFunction* function = m_wasmInternalFunctions[functionIndex].get();
             if (!m_wasmInternalFunctionLinkBuffers[functionIndex]) {
-                Base::fail(makeString("Out of executable memory in function at index ", String::number(functionIndex)));
+                Base::fail(makeString("Out of executable memory in function at index "_s, String::number(functionIndex)));
```
Should say `functionIndex` instead of `String::number(functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmBBQPlan.cpp:
```diff
@@ -322,7 +322,7 @@ void BBQPlan::didCompleteCompilation()
 
                 if (linkBuffer.didFailToAllocate()) {
-                    Base::fail(makeString("Out of executable memory in function entrypoint at index ", String::number(functionIndex)));
+                    Base::fail(makeString("Out of executable memory in function entrypoint at index "_s, String::number(functionIndex)));
```
Should say `functionIndex` instead of `String::number(functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmIndexOrName.cpp:
```diff
@@ -56,8 +56,8 @@ String makeString(const IndexOrName& ion)
     if (ion.isIndex())
-        return makeString(moduleName, ".wasm-function[", String::number(ion.index()), ']');
-    return makeString(moduleName, ".wasm-function[", String(ion.name()->data(), ion.name()->size()), ']');
+        return makeString(moduleName, ".wasm-function["_s, String::number(ion.index()), ']');
```
Should say `ion.index()` instead of `String::number(ion.index())`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmIndexOrName.cpp:
```diff
@@ -56,8 +56,8 @@ String makeString(const IndexOrName& ion)
-        return makeString(moduleName, ".wasm-function[", String::number(ion.index()), ']');
-    return makeString(moduleName, ".wasm-function[", String(ion.name()->data(), ion.name()->size()), ']');
+        return makeString(moduleName, ".wasm-function["_s, String::number(ion.index()), ']');
+    return makeString(moduleName, ".wasm-function["_s, String(ion.name()->data(), ion.name()->size()), ']');
```
Should use `StringView` instead of `String` to accomplish the same thing without allocating and deallocating a temporary string.

darinadler commented on Source/JavaScriptCore/wasm/WasmLLIntPlan.cpp:
```diff
@@ -92,7 +92,7 @@ void LLIntPlan::compileFunction(uint32_t functionIndex)
         if (!m_errorMessage) {
             // Multiple compiles could fail simultaneously. We arbitrarily choose the first.
-            fail(makeString(parseAndCompileResult.error(), ", in function at index ", String::number(functionIndex))); // FIXME make this an Expected.
+            fail(makeString(parseAndCompileResult.error(), ", in function at index "_s, String::number(functionIndex))); // FIXME make this an Expected.
```
Should say `functionIndex` instead of `String::number(functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmLLIntPlan.cpp:
```diff
@@ -174,7 +174,7 @@ void LLIntPlan::didCompleteCompilation()
             LinkBuffer linkBuffer(jit, nullptr, LinkBuffer::Profile::Wasm, JITCompilationCanFail);
             if (UNLIKELY(linkBuffer.didFailToAllocate())) {
-                Base::fail(makeString("Out of executable memory in function entrypoint at index ", String::number(functionIndex)));
+                Base::fail(makeString("Out of executable memory in function entrypoint at index "_s, String::number(functionIndex)));
```
Should say `functionIndex` instead of `String::number(functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmOMGPlan.cpp:
```diff
@@ -103,15 +103,15 @@ void OMGPlan::work(CompilationEffort)
     if (UNLIKELY(!parseAndCompileResult)) {
         Locker locker { m_lock };
-        fail(makeString(parseAndCompileResult.error(), "when trying to tier up ", String::number(m_functionIndex)));
+        fail(makeString(parseAndCompileResult.error(), "when trying to tier up "_s, String::number(m_functionIndex)));
```
Should say `m_functionIndex` instead of `String::number(m_functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmOMGPlan.cpp:
```diff
@@ -103,15 +103,15 @@ void OMGPlan::work(CompilationEffort)
     if (UNLIKELY(linkBuffer.didFailToAllocate())) {
         Locker locker { m_lock };
-        Base::fail(makeString("Out of executable memory while tiering up function at index ", String::number(m_functionIndex)));
+        Base::fail(makeString("Out of executable memory while tiering up function at index "_s, String::number(m_functionIndex)));
```
Should say `m_functionIndex` instead of `String::number(m_functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmOSREntryPlan.cpp:
```diff
@@ -107,15 +107,15 @@ void OSREntryPlan::work(CompilationEffort)
     if (UNLIKELY(!parseAndCompileResult)) {
         Locker locker { m_lock };
-        fail(makeString(parseAndCompileResult.error(), "when trying to tier up ", String::number(m_functionIndex)));
+        fail(makeString(parseAndCompileResult.error(), "when trying to tier up "_s, String::number(m_functionIndex)));
```
Should say `m_functionIndex` instead of `String::number(m_functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/wasm/WasmOSREntryPlan.cpp:
```diff
@@ -107,15 +107,15 @@ void OSREntryPlan::work(CompilationEffort)
     if (UNLIKELY(linkBuffer.didFailToAllocate())) {
         Locker locker { m_lock };
-        Base::fail(makeString("Out of executable memory while tiering up function at index ", String::number(m_functionIndex)));
+        Base::fail(makeString("Out of executable memory while tiering up function at index "_s, String::number(m_functionIndex)));
```
Should say `m_functionIndex` instead of `String::number(m_functionIndex)`. Doesn’t need to be addressed in this patch, but I probably would.

darinadler commented on Source/JavaScriptCore/runtime/AbstractModuleRecord.cpp:
```diff
@@ -764,11 +764,11 @@ JSModuleNamespaceObject* AbstractModuleRecord::getModuleNamespace(JSGlobalObject
         switch (resolution.type) {
         case Resolution::Type::NotFound:
-            throwSyntaxError(globalObject, scope, makeString("Exported binding name '", String(name.get()), "' is not found."));
+            throwSyntaxError(globalObject, scope, makeString("Exported binding name '"_s, String(name.get()), "' is not found."_s));
```
Should be able to pass `*name` or `name.get()` without calling `String` and avoid the reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/AbstractModuleRecord.cpp:
```diff
@@ -858,7 +858,7 @@ static String printableName(const RefPtr<UniquedStringImpl>& uid)
     if (uid->isSymbol())
         return uid.get();
-    return WTF::makeString("'", String(uid.get()), "'");
+    return WTF::makeString('\'', String(uid.get()), '\'');
```
Should be able to pass `*uid` or `uid.get()` without calling `String` and avoid the reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/JSModuleRecord.cpp:
```diff
@@ -126,15 +126,15 @@ void JSModuleRecord::instantiateDeclarations(JSGlobalObject* globalObject, Modul
             switch (resolution.type) {
             case Resolution::Type::NotFound:
-                throwSyntaxError(globalObject, scope, makeString("Indirectly exported binding name '", String(exportEntry.exportName.impl()), "' is not found."));
+                throwSyntaxError(globalObject, scope, makeString("Indirectly exported binding name '"_s, String(exportEntry.exportName.impl()), "' is not found."_s));
```
Should be able to pass `exportEntry.exportName` without `String` and avoid the reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/JSModuleRecord.cpp:
```diff
@@ -126,15 +126,15 @@ void JSModuleRecord::instantiateDeclarations(JSGlobalObject* globalObject, Modul
 
             case Resolution::Type::Ambiguous:
-                throwSyntaxError(globalObject, scope, makeString("Indirectly exported binding name '", String(exportEntry.exportName.impl()), "' cannot be resolved due to ambiguous multiple bindings."));
+                throwSyntaxError(globalObject, scope, makeString("Indirectly exported binding name '"_s, String(exportEntry.exportName.impl()), "' cannot be resolved due to ambiguous multiple bindings."_s));
```
Should be able to pass `exportEntry.exportName` without `String` and avoid the reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/JSModuleRecord.cpp:
```diff
@@ -126,15 +126,15 @@ void JSModuleRecord::instantiateDeclarations(JSGlobalObject* globalObject, Modul
 
             case Resolution::Type::Error:
-                throwSyntaxError(globalObject, scope, makeString("Indirectly exported binding name 'default' cannot be resolved by star export entries."));
+                throwSyntaxError(globalObject, scope, makeString("Indirectly exported binding name 'default' cannot be resolved by star export entries."_s));
```
Should not need to call `makeString` on a single `ASCIILiteral`.

darinadler commented on Source/JavaScriptCore/runtime/JSModuleRecord.cpp:
```diff
@@ -167,15 +167,15 @@ void JSModuleRecord::instantiateDeclarations(JSGlobalObject* globalObject, Modul
             switch (resolution.type) {
             case Resolution::Type::NotFound:
-                throwSyntaxError(globalObject, scope, makeString("Importing binding name '", String(importEntry.importName.impl()), "' is not found."));
+                throwSyntaxError(globalObject, scope, makeString("Importing binding name '"_s, String(importEntry.importName.impl()), "' is not found."_s));
```
Should be able to pass `importEntry.importName` without `String` and avoid the reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/JSModuleRecord.cpp:
```diff
@@ -167,15 +167,15 @@ void JSModuleRecord::instantiateDeclarations(JSGlobalObject* globalObject, Modul
 
             case Resolution::Type::Ambiguous:
-                throwSyntaxError(globalObject, scope, makeString("Importing binding name '", String(importEntry.importName.impl()), "' cannot be resolved due to ambiguous multiple bindings."));
+                throwSyntaxError(globalObject, scope, makeString("Importing binding name '"_s, String(importEntry.importName.impl()), "' cannot be resolved due to ambiguous multiple bindings."_s));
```
Should be able to pass `importEntry.importName` without `String` and avoid the reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/JSModuleRecord.cpp:
```diff
@@ -167,15 +167,15 @@ void JSModuleRecord::instantiateDeclarations(JSGlobalObject* globalObject, Modul
 
             case Resolution::Type::Error:
-                throwSyntaxError(globalObject, scope, makeString("Importing binding name 'default' cannot be resolved by star export entries."));
+                throwSyntaxError(globalObject, scope, makeString("Importing binding name 'default' cannot be resolved by star export entries."_s));
```
Should not need to call `makeString` on a single `ASCIILiteral`.

darinadler commented on Source/JavaScriptCore/runtime/Options.cpp:
```diff
@@ -1020,7 +1020,7 @@ bool Options::setAliasedOption(const char* arg, bool verify)
             if (!invertedValueStr)                                      \
                 return false;                                           \
-            unaliasedOption = unaliasedOption + "=" + invertedValueStr; \
+            unaliasedOption = unaliasedOption + "="_s + invertedValueStr; \
```
Might want to use `makeString` here.

darinadler commented on Source/JavaScriptCore/runtime/ProgramExecutable.cpp:
```diff
@@ -101,7 +101,7 @@ JSObject* ProgramExecutable::initializeGlobalProperties(VM& vm, JSGlobalObject*
         for (auto& entry : lexicalDeclarations) {
             if (variableDeclarations.contains(entry.key))
-                return createSyntaxError(globalObject, makeString("Can't create duplicate variable: '", String(entry.key.get()), "'"));
+                return createSyntaxError(globalObject, makeString("Can't create duplicate variable: '"_s, String(entry.key.get()), '\''));
```
Should be able to pass `entry.key` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/ProgramExecutable.cpp:
```diff
@@ -112,7 +112,7 @@ JSObject* ProgramExecutable::initializeGlobalProperties(VM& vm, JSGlobalObject*
             switch (status) {
             case GlobalPropertyLookUpStatus::NonConfigurable:
-                return createSyntaxError(globalObject, makeString("Can't create duplicate variable that shadows a global property: '", String(entry.key.get()), "'"));
+                return createSyntaxError(globalObject, makeString("Can't create duplicate variable that shadows a global property: '"_s, String(entry.key.get()), '\''));
```
Should be able to pass `entry.key` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/ProgramExecutable.cpp:
```diff
@@ -134,7 +134,7 @@ JSObject* ProgramExecutable::initializeGlobalProperties(VM& vm, JSGlobalObject*
                         continue;
                 }
-                return createSyntaxError(globalObject, makeString("Can't create duplicate variable: '", String(entry.key.get()), "'"));
+                return createSyntaxError(globalObject, makeString("Can't create duplicate variable: '"_s, String(entry.key.get()), '\''));
```
Should be able to pass `entry.key` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/ProgramExecutable.cpp:
```diff
@@ -145,7 +145,7 @@ JSObject* ProgramExecutable::initializeGlobalProperties(VM& vm, JSGlobalObject*
                 RETURN_IF_EXCEPTION(throwScope, nullptr);
                 if (hasProperty)
-                    return createSyntaxError(globalObject, makeString("Can't create duplicate variable: '", String(entry.key.get()), "'"));
+                    return createSyntaxError(globalObject, makeString("Can't create duplicate variable: '"_s, String(entry.key.get()), '\''));
```
Should be able to pass `entry.key` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/ProxyObject.cpp:
```diff
@@ -436,7 +436,7 @@ bool ProxyObject::performPut(JSGlobalObject* globalObject, JSValue putValue, JSV
     if (!trapResultAsBool) {
         if (shouldThrow)
-            throwTypeError(globalObject, scope, makeString("Proxy object's 'set' trap returned falsy value for property '", String(propertyName.uid()), "'"));
+            throwTypeError(globalObject, scope, makeString("Proxy object's 'set' trap returned falsy value for property '"_s, String(propertyName.uid()), '\''));
```
Should be able to pass `propertyName.uid()` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/ProxyObject.cpp:
```diff
@@ -834,7 +834,7 @@ bool ProxyObject::performDefineOwnProperty(JSGlobalObject* globalObject, Propert
     if (!trapResultAsBool) {
         if (shouldThrow)
-            throwTypeError(globalObject, scope, makeString("Proxy's 'defineProperty' trap returned falsy value for property '", String(propertyName.uid()), "'"));
+            throwTypeError(globalObject, scope, makeString("Proxy's 'defineProperty' trap returned falsy value for property '"_s, String(propertyName.uid()), '\''));
```
Should be able to pass `propertyName.uid()` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/ProxyObject.cpp:
```diff
@@ -967,15 +967,15 @@ void ProxyObject::performGetOwnPropertyNames(JSGlobalObject* globalObject, Prope
     for (UniquedStringImpl* impl : targetNonConfigurableKeys) {
         if (!uncheckedResultKeys.remove(impl)) {
-            throwTypeError(globalObject, scope, makeString("Proxy object's 'target' has the non-configurable property '", String(impl), "' that was not in the result from the 'ownKeys' trap"));
+            throwTypeError(globalObject, scope, makeString("Proxy object's 'target' has the non-configurable property '"_s, String(impl), "' that was not in the result from the 'ownKeys' trap"_s));
```
Should be able to pass `impl` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/ProxyObject.cpp:
```diff
@@ -967,15 +967,15 @@ void ProxyObject::performGetOwnPropertyNames(JSGlobalObject* globalObject, Prope
         for (UniquedStringImpl* impl : targetConfigurableKeys) {
             if (!uncheckedResultKeys.remove(impl)) {
-                throwTypeError(globalObject, scope, makeString("Proxy object's non-extensible 'target' has configurable property '", String(impl), "' that was not in the result from the 'ownKeys' trap"));
+                throwTypeError(globalObject, scope, makeString("Proxy object's non-extensible 'target' has configurable property '"_s, String(impl), "' that was not in the result from the 'ownKeys' trap"_s));
```
Should be able to pass `impl` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/runtime/Symbol.cpp:
```diff
@@ -85,7 +85,7 @@ void Symbol::destroy(JSCell* cell)
 String Symbol::descriptiveString() const
 {
-    return makeString("Symbol(", String(m_privateName.uid()), ')');
+    return makeString("Symbol("_s, String(m_privateName.uid()), ')');
```
Should be able to pass `m_privateName.uid()` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/wasm/WasmEntryPlan.cpp:
```diff
@@ -132,7 +132,7 @@ void EntryPlan::prepare()
             case BindingFailure::OutOfMemory: {
                 Locker locker { m_lock };
-                return fail(makeString("Out of executable memory at import ", String::number(importIndex)));
+                return fail(makeString("Out of executable memory at import "_s, String::number(importIndex)));
```
Should be `importIndex` instead of `String::number(importIndex)`.

darinadler commented on Source/JavaScriptCore/wasm/js/WebAssemblyModuleRecord.cpp:
```diff
@@ -155,15 +155,15 @@ void WebAssemblyModuleRecord::initializeImports(JSGlobalObject* globalObject, JS
             switch (resolution.type) {
             case Resolution::Type::NotFound:
-                throwSyntaxError(globalObject, scope, makeString("Importing binding name '", String(fieldName.impl()), "' is not found."));
+                throwSyntaxError(globalObject, scope, makeString("Importing binding name '"_s, String(fieldName.impl()), "' is not found."_s));
```
Should be able to pass `fieldName` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/wasm/js/WebAssemblyModuleRecord.cpp:
```diff
@@ -155,15 +155,15 @@ void WebAssemblyModuleRecord::initializeImports(JSGlobalObject* globalObject, JS
 
             case Resolution::Type::Ambiguous:
-                throwSyntaxError(globalObject, scope, makeString("Importing binding name '", String(fieldName.impl()), "' cannot be resolved due to ambiguous multiple bindings."));
+                throwSyntaxError(globalObject, scope, makeString("Importing binding name '"_s, String(fieldName.impl()), "' cannot be resolved due to ambiguous multiple bindings."_s));
```
Should be able to pass `fieldName` without `String` to avoid reference count churn.

darinadler commented on Source/JavaScriptCore/wasm/js/WebAssemblyModuleRecord.cpp:
```diff
@@ -155,15 +155,15 @@ void WebAssemblyModuleRecord::initializeImports(JSGlobalObject* globalObject, JS
 
             case Resolution::Type::Error:
-                throwSyntaxError(globalObject, scope, makeString("Importing binding name 'default' cannot be resolved by star export entries."));
+                throwSyntaxError(globalObject, scope, makeString("Importing binding name 'default' cannot be resolved by star export entries."_s));
```
Should be able to pass an `ASCIILiteral` without `makeString`.

darinadler commented on Source/JavaScriptCore/API/glib/JSCContext.cpp:
```diff
@@ -294,7 +294,7 @@ JSValueRef jscContextGArrayToJSArray(JSCContext* context, GPtrArray* gArray, JSV
             JSObjectSetPropertyAtIndex(priv->jsContext.get(), jsArrayObject, i, jscValueGetJSValue(JSC_VALUE(item)), exception);
         else
-            *exception = toRef(JSC::createTypeError(globalObject, makeString("invalid item type in GPtrArray")));
+            *exception = toRef(JSC::createTypeError(globalObject, makeString("invalid item type in GPtrArray"_s)));
```
`makeString` is not needed when it’s just a single argument. This occurs here and a couple other places.

zdobersek replied:
Found, fixed.

zdobersek committed:
Adopt ASCIILiterals in makeString, tryMakeString, operator+ calls inside JavaScriptCore
https://bugs.webkit.org/show_bug.cgi?id=252167

Reviewed by Darin Adler.

Convert different cases of makeString(), tryMakeString() and StringAppend-producing
operator+ calls from using native C string literals to using more-desired ASCIILiterals.

* Source/JavaScriptCore/API/glib/JSCCallbackFunction.cpp:
(JSC::JSCCallbackFunction::construct):
* Source/JavaScriptCore/API/glib/JSCContext.cpp:
(jscContextGArrayToJSArray):
(jscContextJSArrayToGArray):
(jscContextGValueToJSValue):
(jscContextJSValueToGValue):
* Source/JavaScriptCore/API/glib/JSCValue.cpp:
(jsc_value_new_array):
(jscValueCallFunction):
* Source/JavaScriptCore/API/tests/ExecutionTimeLimitTest.cpp:
(testExecutionTimeLimit):
* Source/JavaScriptCore/API/tests/testapi.cpp:
(TestAPI::markedJSValueArrayAndGC):
* Source/JavaScriptCore/assembler/LinkBuffer.cpp:
(JSC::LinkBuffer::linkComments):
* Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp:
(JSC::DFG::SpeculativeJIT::runSlowPathGenerators):
(JSC::DFG::SpeculativeJIT::compileCurrentBlock):
* Source/JavaScriptCore/ftl/FTLLowerDFGToB3.cpp:
(JSC::FTL::DFG::LowerDFGToB3::compileCompareStrictEq):
* Source/JavaScriptCore/inspector/IdentifiersFactory.cpp:
(Inspector::addPrefixToIdentifier):
* Source/JavaScriptCore/inspector/InjectedScriptBase.cpp:
(Inspector::InjectedScriptBase::makeCall):
(Inspector::InjectedScriptBase::makeAsyncCall):
* Source/JavaScriptCore/inspector/InspectorBackendDispatcher.cpp:
(Inspector::BackendDispatcher::dispatch):
(Inspector::BackendDispatcher::getPropertyValue):
* Source/JavaScriptCore/inspector/JSGlobalObjectConsoleClient.cpp:
(Inspector::JSGlobalObjectConsoleClient::profile):
(Inspector::JSGlobalObjectConsoleClient::profileEnd):
(Inspector::JSGlobalObjectConsoleClient::warnUnimplemented):
* Source/JavaScriptCore/inspector/agents/InspectorAuditAgent.cpp:
(Inspector::InspectorAuditAgent::run):
* Source/JavaScriptCore/inspector/agents/InspectorConsoleAgent.cpp:
(Inspector::InspectorConsoleAgent::enable):
(Inspector::InspectorConsoleAgent::startTiming):
(Inspector::InspectorConsoleAgent::logTiming):
(Inspector::InspectorConsoleAgent::stopTiming):
(Inspector::InspectorConsoleAgent::count):
(Inspector::InspectorConsoleAgent::countReset):
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp:
(Inspector::objectGroupForBreakpointAction):
(Inspector::InspectorDebuggerAgent::ProtocolBreakpoint::ProtocolBreakpoint):
* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::Interpreter::executeEval):
* Source/JavaScriptCore/jit/JIT.cpp:
(JSC::JIT::privateCompileMainPass):
(JSC::JIT::privateCompileSlowCases):
* Source/JavaScriptCore/jsc.cpp:
(GlobalObject::moduleLoaderImportModule):
(GlobalObject::moduleLoaderResolve):
(GlobalObject::moduleLoaderFetch):
(FunctionJSCStackFunctor::operator() const):
(computeFilePath):
* Source/JavaScriptCore/parser/Lexer.cpp:
(JSC::Lexer<T>::invalidCharacterMessage const):
(JSC::Lexer<LChar>::parseIdentifier):
(JSC::Lexer<T>::scanRegExp):
* Source/JavaScriptCore/parser/Parser.cpp:
(JSC::Parser<LexerType>::parseInner):
* Source/JavaScriptCore/parser/Parser.h:
(JSC::Parser::updateErrorWithNameAndMessage):
* Source/JavaScriptCore/runtime/AbstractModuleRecord.cpp:
(JSC::AbstractModuleRecord::getModuleNamespace):
(JSC::printableName):
* Source/JavaScriptCore/runtime/BytecodeCacheError.cpp:
(JSC::BytecodeCacheError::WriteError::message const):
* Source/JavaScriptCore/runtime/Error.cpp:
(JSC::makeDOMAttributeGetterTypeErrorMessage):
(JSC::makeDOMAttributeSetterTypeErrorMessage):
(JSC::throwConstructorCannotBeCalledAsFunctionTypeError):
(JSC::createOutOfMemoryError):
* Source/JavaScriptCore/runtime/ErrorInstance.cpp:
(JSC::ErrorInstance::sanitizedToString):
* Source/JavaScriptCore/runtime/ExceptionHelpers.cpp:
(JSC::createUndefinedVariableError):
(JSC::defaultApproximateSourceError):
(JSC::defaultSourceAppender):
(JSC::invalidParameterInSourceAppender):
(JSC::invalidParameterInstanceofSourceAppender):
(JSC::invalidPrototypeSourceAppender):
(JSC::createErrorForInvalidGlobalAssignment):
* Source/JavaScriptCore/runtime/FunctionConstructor.cpp:
(JSC::stringifyFunction):
* Source/JavaScriptCore/runtime/IntlCollator.cpp:
(JSC::IntlCollator::initializeCollator):
* Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
(JSC::IntlDateTimeFormat::initializeDateTimeFormat):
* Source/JavaScriptCore/runtime/IntlDurationFormat.cpp:
(JSC::IntlDurationFormat::initializeDurationFormat):
* Source/JavaScriptCore/runtime/IntlNumberFormat.cpp:
(JSC::IntlNumberFormat::initializeNumberFormat):
* Source/JavaScriptCore/runtime/IntlObject.cpp:
(JSC::intlDefaultNumberOption):
(JSC::canonicalizeLocaleList):
* Source/JavaScriptCore/runtime/IntlRelativeTimeFormat.cpp:
(JSC::IntlRelativeTimeFormat::initializeRelativeTimeFormat):
* Source/JavaScriptCore/runtime/JSCJSValueInlines.h:
(JSC::JSValue::toIndex const):
(JSC::JSValue::toTypedArrayIndex const):
* Source/JavaScriptCore/runtime/JSCustomGetterFunction.cpp:
(JSC::JSCustomGetterFunction::create):
* Source/JavaScriptCore/runtime/JSCustomSetterFunction.cpp:
(JSC::JSCustomSetterFunction::create):
* Source/JavaScriptCore/runtime/JSFunction.cpp:
(JSC::makeNameWithOutOfMemoryCheck):
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::init):
* Source/JavaScriptCore/runtime/JSModuleLoader.cpp:
(JSC::JSModuleLoader::importModule):
(JSC::JSModuleLoader::fetch):
* Source/JavaScriptCore/runtime/JSModuleRecord.cpp:
(JSC::JSModuleRecord::instantiateDeclarations):
* Source/JavaScriptCore/runtime/JSObject.cpp:
(JSC::JSObject::putDirectNativeIntrinsicGetter):
(JSC::JSObject::putDirectNativeIntrinsicGetterWithoutTransition):
* Source/JavaScriptCore/runtime/LiteralParser.cpp:
(JSC::LiteralParser<CharType>::Lexer::lex):
(JSC::LiteralParser<CharType>::Lexer::lexStringSlow):
(JSC::LiteralParser<CharType>::parsePrimitiveValue):
* Source/JavaScriptCore/runtime/LiteralParser.h:
(JSC::LiteralParser::getErrorMessage):
* Source/JavaScriptCore/runtime/NumberPrototype.cpp:
(JSC::throwVMToThisNumberError):
* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::Options::setAliasedOption):
* Source/JavaScriptCore/runtime/ProgramExecutable.cpp:
(JSC::ProgramExecutable::initializeGlobalProperties):
* Source/JavaScriptCore/runtime/ProxyObject.cpp:
(JSC::ProxyObject::performPut):
(JSC::ProxyObject::performDefineOwnProperty):
(JSC::ProxyObject::performGetOwnPropertyNames):
* Source/JavaScriptCore/runtime/SamplingProfiler.cpp:
(JSC::SamplingProfiler::reportTopBytecodes):
* Source/JavaScriptCore/runtime/Symbol.cpp:
(JSC::Symbol::descriptiveString const):
* Source/JavaScriptCore/wasm/WasmBBQPlan.cpp:
(JSC::Wasm::BBQPlan::work):
(JSC::Wasm::BBQPlan::compileFunction):
(JSC::Wasm::BBQPlan::didCompleteCompilation):
* Source/JavaScriptCore/wasm/WasmEntryPlan.cpp:
(JSC::Wasm::EntryPlan::prepare):
* Source/JavaScriptCore/wasm/WasmEntryPlan.h:
(JSC::Wasm::EntryPlan::tryReserveCapacity):
* Source/JavaScriptCore/wasm/WasmIndexOrName.cpp:
(JSC::Wasm::makeString):
* Source/JavaScriptCore/wasm/WasmLLIntPlan.cpp:
(JSC::Wasm::LLIntPlan::compileFunction):
(JSC::Wasm::LLIntPlan::didCompleteCompilation):
* Source/JavaScriptCore/wasm/WasmOMGPlan.cpp:
(JSC::Wasm::OMGPlan::work):
* Source/JavaScriptCore/wasm/WasmOSREntryPlan.cpp:
(JSC::Wasm::OSREntryPlan::work):
* Source/JavaScriptCore/wasm/js/WebAssemblyExceptionPrototype.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/wasm/js/WebAssemblyModuleRecord.cpp:
(JSC::WebAssemblyModuleRecord::initializeImports):

Canonical link: https://commits.webkit.org/261376@main
